Std/Data/List/Lemmas.lean

```diff
@@ -1763,6 +1763,19 @@ theorem disjoint_take_drop : ∀ {l : List α}, l.Nodup → m ≤ n → Disjoint
       refine ⟨fun h => h₀ _ (mem_of_mem_drop h) rfl, ?_⟩
       exact disjoint_take_drop h₁ (Nat.le_of_succ_le_succ h)
 
+theorem drop_eq_nil_of_eq_nil {as : List α} {i : Nat} (h: as = []) : as.drop i = [] :=
+  List.drop_eq_nil_of_le <| h ▸ List.length_nil ▸ zero_le i
```

Comment:
Simpler still:
```lean
theorem drop_eq_nil_of_eq_nil : ∀ {as : List α} {i}, as = [] → as.drop i = []
  | _, _, rfl => drop_nil
```
Same could be done for `take_eq_nil_of_eq_nil` after adding the missing `take_nil` lemma.

Reply:
Great! I also like the other golf with `bt`.
I added the `take_nil` lemma and was about to open a pr, but then I read the [contribution guidelines](https://github.com/leanprover/lean4/blob/c268d7e97bb075427a8412cceecc4e483379ac25/CONTRIBUTING.md)
and now I'm not sure if I should ask on zulip first. I think it will probably take 1minute of the reviewers time, but still.. What do you think?

https://github.com/mo271/lean4/commit/05273e9eed5f3680f5e2f199506b76f3764d256f

Reply:
[Converting to draft, since the current way would depend on the propsed `take_nil` lemma.

Reply:
Those are guidelines for lean4 core not for std4! Just add `take_nil` in this PR and it should be fine.

Reply:
Yes, I just thought it would be nice to have `take_nil` in the same place as `drop_nil`, but I guess we can also put it in std4...

Reply:
Added `take_nil` as suggested.

Reply:
Adding things to core that aren't actually needed in core is a no-go... Who knows, maybe `drop_nil` will eventually be moved to std4!